ttnn/cpp/ttnn/operations/matmul/device/matmul_op_multi_core_reuse_mcast_dram_sharded_program_factory.cpp

```diff
@@ -310,6 +310,92 @@ void get_dram_reader_core_coords_wormhole_b0(
     all_cores_ordered = adj_core_logical_realloc;
 }
 
+void get_dram_reader_core_coords_blackhole(
```

Comment:
This function has a lot of common elements with the GS and WH functions.

Do you think it's a good idea to factor out common functionality? If yes, would it be better in this PR or a subsequent PR?

Reply:
yeah that should make code cleaner, I can do in next PR, this one should just focus on unblock the BH